pkg/store/memstore/memstore.go

```diff
@@ -22,30 +28,84 @@ func New() *memoryStore {
 	}
 }
 
-func (s *memoryStore) ReadMetrics(ctx context.Context, minTimestampMs int64, fn func(partitionKey string, families []*clientmodel.MetricFamily) error) error {
-	s.lock.Lock()
-	store := s.store
-	s.store = make(map[string]*clusterMetricSlice)
-	s.lock.Unlock()
+// StartCleaner starts a goroutine, executing the cleanup of stored data
+// at regular intervals specified by "interval".
+// The goroutine will be stopped when the given context is done.
+func (s *memoryStore) StartCleaner(ctx context.Context, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				s.cleanup(time.Now)
+			case <-ctx.Done():
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (s *memoryStore) cleanup(now func() time.Time) {
```

Comment:
why not simply pass a `time.Time`? Or better yet, eliminate the parameter all together? I think this implementation has two issues:
1. Likely to cause confusion because `now` is actually the time when the function is called in the for-loop, not when the `cleanup` func is called
2. It is cumbersome for calling cleanup with a time other than now, as we have to do something like
    ```go
    s.cleanup(func() time.Time {return time.Now().Add(-5*time.Minute)})
    ```
    Instead of something like
    ```go
    s.cleanup(time.Now().Add(-5*time.Minute))
    ```

Additionally, the semantics of this memory-store are clear: it is TTL-based. This has little meaning for TTLs relative to times other than now. In any case, the TTL could be a configurable parameter.

To simplify here, I propose eliminating the `now` parameter entirely.

Reply:
I use `now` only for testing only to pass a different "now" function, there is no other meaning.

The invoking function in non-tests is as simple as:
https://github.com/openshift/telemeter/blob/1dfcc6840e22399707cf64d60111426661a7efcf/pkg/store/memstore/memstore.go#L41.

But I am totally fine with passing `time.Time()` instead.

For testability I would still recommend injecting it so we don't have potential flakes in tests.

Reply:
Ok, then let's do that. Let's make this accept a `time.Time` instead.